From 992a2e98add398e4c081489d747cd62332ed7e2b Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@microsoft.com>
Date: Fri, 4 Oct 2019 14:45:41 +0100
Subject: [PATCH] portabled: allow to detach an image with a unit in
 linked-runtime state

This is necessary when a directory was attached with
--copy=symlink, otherwise detach will always fail.

Fixed #13725

(cherry picked from commit c3d809ef72db616391a1a2b738eae137f9024e3f)
---
 src/portable/portable.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/portable/portable.c b/src/portable/portable.c
index eb5daf3b0e..5fd14c70b6 100644
--- a/src/portable/portable.c
+++ b/src/portable/portable.c
@@ -1185,7 +1185,7 @@ int portable_detach(
                 r = unit_file_lookup_state(UNIT_FILE_SYSTEM, &paths, de->d_name, &state);
                 if (r < 0)
                         return log_debug_errno(r, "Failed to determine unit file state of '%s': %m", de->d_name);
-                if (!IN_SET(state, UNIT_FILE_STATIC, UNIT_FILE_DISABLED, UNIT_FILE_LINKED, UNIT_FILE_RUNTIME))
+                if (!IN_SET(state, UNIT_FILE_STATIC, UNIT_FILE_DISABLED, UNIT_FILE_LINKED, UNIT_FILE_RUNTIME, UNIT_FILE_LINKED_RUNTIME))
                         return sd_bus_error_setf(error, BUS_ERROR_UNIT_EXISTS, "Unit file '%s' is in state '%s', can't detach.", de->d_name, unit_file_state_to_string(state));
 
                 r = unit_file_is_active(bus, de->d_name, error);
-- 
2.20.1

